app/src/main/java/com/kickstarter/viewmodels/NewCardFragmentViewModel.kt

```diff
@@ -111,34 +156,43 @@ interface NewCardFragmentViewModel {
                     .subscribe { this.cardWidgetFocusDrawable.onNext(it) }
 
             val saveCardNotification = cardForm
-                    .compose<CardForm>(takeWhen(this.saveCardClicked))
-                    .map { storeNameAndPostalCode(it) }
+                    .map {
+                        it.card?.let { card ->
+                            card.name = it.name
+                            card.addressZip = it.postalCode
+                            card
+                        }
+                    }
+                    .compose<Pair<Card, Boolean>>(combineLatestPair(reusable))
+                    .compose<Pair<Card, Boolean>>(takeWhen(this.saveCardClicked))
                     .switchMap { createTokenAndSaveCard(it).materialize() }
                     .compose(bindToLifecycle())
                     .share()
 
             saveCardNotification
                     .compose(values())
-                    .subscribe {
-                        this.success.onNext(null)
-                        this.koala.trackSavedPaymentMethod()
-                    }
+                    .subscribe()
 
             saveCardNotification
```

Comment:
Could you move this down close to the `this.koala.trackViewedAddNewCard()` line? This way we can keep the koala related code grouped.